src/IO/H5/CheckH5PropertiesMatch.hpp

```diff
@@ -0,0 +1,26 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace h5 {
+/*!
```

Comment:
Could you please add tests for these two functions? I imagine them writing out a couple files, say 3, where 2 of them have matching observation IDs and the third is missing one or has an extra. I don't think it's easy to insert the wrong source tree, so probably just check that it's the same for all 3 files is fine. You can just do something like 1 element in each file with 2 grid points in 1d.